### DISCUSSION TITLE
TUN模式可以指定进程吗？

### DISCUSSION THREAD

**Discussion by eniacnemesis** [clash-verge-rev/clash-verge-rev#1694] [General]
请问可不可以仅让指定的应用程序走TUN模式，其他的直连？

**Comment by jerry-rs** [comment]
当然可以呀～, 采用process-name ,具体的配置见：https://wiki.metacubex.one/config/rules/#process-name

**Reply by eniacnemesis** [reply]
谢谢，我先看看。

**Reply by eniacnemesis** [reply]
我想请教一下
我的目的是只让某个或者某几个无法使用系统代理的应用程序走TUN模式，但不想让所有无法使用系统代理的应用程序都走TUN模式。
我在TUN模式设置中关闭了自动全局路由：
![image](https://github.com/user-attachments/assets/d0eb9061-af0f-4ad7-8300-72f990c591d0)
然后在规则中配置了process-name使用代理的前置规则：
![image](https://github.com/user-attachments/assets/951ab2e0-9b7e-45ee-bd74-12719fd4b92e)

这样配置的结果是除了规则指定的应用程序外其他的应用程序似乎无法访问网络，规则指定的应用程序也时灵时不灵，比如昨晚上试过还是可以的，今天早上就不行了。
是配置有什么问题吗？

**Reply by mengdegege** [reply]
在路由阶段，安卓有这种配置可以匹配包名，Linux可能可以通过IPtables 间接实现，Windows上面貌似做不到， tun 只支持对路由做配置，指定哪些ip段走tun。 process-name是规则配置里面的东西，都匹配规则了就已经进入tun了。

不过，如果只是希望通过进程名来匹配规则进行出口选择，倒是可以。
[PROCESS-PATH](https://wiki.metacubex.one/config/rules/#process-path)
[PROCESS-PATH-REGEX](https://wiki.metacubex.one/config/rules/#process-path-regex)
[PROCESS-NAME](https://wiki.metacubex.one/config/rules/#process-name)
[PROCESS-NAME-REGEX](https://wiki.metacubex.one/config/rules/#process-name-regex)
这4个参数都可以做到。

**Comment by kkyimengweima** [comment]
make
